src/main/java/org/prebid/server/bidder/adquery/AdQueryBidder.java

```diff
@@ -0,0 +1,213 @@
+        final Integer h = banner.getH();
+        if (w != null && h != null) {
+            return "%sx%s".formatted(w, h);
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private int getIntOrElseZero(Integer number) {
+        return number != null ? number : 0;
+    }
+
+    private MultiMap resolveHeader(Device device) {
+        final MultiMap headers = HttpUtil.headers();
+        headers.add(HttpUtil.X_OPENRTB_VERSION_HEADER, ORTB_VERSION);
+
+        if (Objects.nonNull(device) && StringUtils.isNotBlank(device.getIp()) && device.getIp().length() > 0) {
+            headers.add(HttpUtil.X_FORWARDED_FOR_HEADER, device.getIp());
+        }
+
+        return headers;
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(BidderCall<AdQueryRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final AdQueryResponse bidResponse = mapper.decodeValue(
+                    httpCall.getResponse().getBody(), AdQueryResponse.class);
+            return Result.withValues(extractBids(bidResponse, bidRequest));
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(AdQueryResponse adQueryResponse, BidRequest bidRequest) {
+        if (adQueryResponse == null || Objects.isNull(adQueryResponse.getData())) {
+            return Collections.emptyList();
+        }
+
+        final AdQueryDataResponse data = adQueryResponse.getData();
+        final Bid bid = Bid.builder()
+                .id(data.getRequestId())
+                .impid(resoleImpId(bidRequest, data))
+                .price(BigDecimalParser.parse(data.getCpm()))
+                .adm(String.format(ADM_TEMPLATE, data.getAdqLib(), data.getTag()))
+                .adomain(data.getAdDomains())
+                .crid(String.format("%d", data.getCreationId()))
+                .w(parseMeasure(ObjectUtil.getIfNotNull(data.getAdQueryMediaType(), AdQueryMediaType::getWidth)))
+                .h(parseMeasure(ObjectUtil.getIfNotNull(data.getAdQueryMediaType(), AdQueryMediaType::getHeight)))
+                .build();
+
+        return Collections.singletonList(BidderBid.of(bid, resolveMediaType(data),
+                StringUtils.isNotBlank(data.getCurrency()) ? data.getCurrency() : DEFAULT_CURRENCY));
+    }
+
+    private static String resoleImpId(BidRequest bidRequest, AdQueryDataResponse data) {
+        return Objects.nonNull(data.getRequestId())
+                ? bidRequest.getId().replaceAll(data.getRequestId(), StringUtils.EMPTY)
+                : bidRequest.getId();
+    }
+
+    private static Integer parseMeasure(String measure) {
+        try {
+            return Integer.valueOf(measure);
+        } catch (NumberFormatException e) {
+            throw new PreBidException("Value of measure: %s can not be parsed.".formatted(measure));
+        }
```

Comment:
parsing can be done by the jackson on deserialization step, just expect Integer type in the DTO, thus this method is redundant